worlds/AutoSNIClient.py

```diff
@@ -27,6 +45,9 @@ async def get_handler(ctx: SNIContext) -> Optional[SNIClient]:
 
 class SNIClient(abc.ABC, metaclass=AutoSNIClientRegister):
 
+    patch_suffix: ClassVar[Optional[Union[str, Tuple[str, ...]]]]
```

Comment:
I think it would be better to remove `Optional` so that there are fewer special cases to handle.

If someone wants to define `patch_suffix` with nothing in it, they can use empty list or empty tuple. (Not defining it at all is also an option.)

Also, putting this type annotation will tell the type checker that this is always present. So if a subclass doesn't define it and then someone follow that information that it's always present, it will crash. To avoid that, we can define a default here of an empty tuple.

Also, you've specified `Tuple` as the container type, but then used `list`s in the games.
`typing.Iterable` would be an appropriate annotation that works for both `tuple` and `list`

So with all of these changes, this line would be:
```python
    patch_suffix: ClassVar[Union[str, Iterable[str]]] = ()
```

You could remove the `if dct["patch_suffix"] is not None:` above.